### PR TITLE
fix terraform destroy for k8s

### PR DESCRIPTION
**Pull request by MilanCorreggio**

### CODE REVIEW

File: test/src/integration/integration_test.go [hunk 1]
```diff
@@ -81,13 +81,18 @@ func TestEndToEndK8s(t *testing.T) {
 
 		scalarModules := []string{"kubernetes", "cassandra", "network"}
 
+
```
Comment (feeblefakie) [inline test/src/integration/integration_test.go:84]: Why 2 empty lines are added?

File: test/src/integration/integration_test.go [hunk 2]
```diff
@@ -81,13 +81,18 @@ func TestEndToEndK8s(t *testing.T) {
...
 		for _, m := range scalarModules {
 			terraformOptions := &terraform.Options{
 				TerraformDir: *terraformDir + *cloudProvider + "/" + m,
 				Vars:         map[string]interface{}{},
 				NoColor:      true,
 			}
-
+			if m == "kubernetes" {
+				terraform.RunTerraformCommand(t, terraformOptions, "init")
+				terraform.RunTerraformCommand(t, terraformOptions, "state", "rm", "module.kubernetes.kubernetes_config_map.aws_auth")
```
Comment (feeblefakie) [inline test/src/integration/integration_test.go:94]: Have you done gofmt?

**Review by feeblefakie — COMMENTED** [review]
Left some comments.

File: test/src/integration/integration_test.go (hunk 2 above)
Comment (MilanCorreggio) [reply test/src/integration/integration_test.go:94]: No I did not know about this.
Thank you I will take a look.

File: test/src/integration/integration_test.go [hunk 3]
```diff
@@ -81,13 +81,18 @@ func TestEndToEndK8s(t *testing.T) {
...
+
 		for _, m := range scalarModules {
 			terraformOptions := &terraform.Options{
 				TerraformDir: *terraformDir + *cloudProvider + "/" + m,
 				Vars:         map[string]interface{}{},
 				NoColor:      true,
 			}
-
+			if m == "kubernetes" {
+				terraform.RunTerraformCommand(t, terraformOptions, "init")
```
Comment (miseyu) [inline test/src/integration/integration_test.go:93]: I don't think it's necessary here since I think you've already done the `terraform init` process.
https://github.com/scalar-labs/scalar-terratest/blob/ecbf1fc3056574cd8763e5e86dd8d5f0599823cf/test/src/integration/integration_ test.go#L114

Comment (MilanCorreggio) [reply test/src/integration/integration_test.go:93]: I have done some test and actually it is.
Before my commit `adding terraform init` the test where failing because it could not find state file.

Comment (miseyu) [reply test/src/integration/integration_test.go:93]: Thank you very much for your understanding. :bow:

**Review by miseyu — APPROVED** [review]
LGTM

File: test/src/integration/integration_test.go (hunk 3 above)
Comment (feeblefakie) [reply test/src/integration/integration_test.go:91]: Do we need `init` here? or we don't need it.
If we need it, it is weird since it's already initialized as Yuji pointed out, so can you clarify? (why do we need it?)

The goal of this is to understand what is going on to fix the root cause, not to make it somehow work.

**Review by miseyu — APPROVED** [review]

File: test/src/integration/integration_test.go (hunk 3 above)
Comment (feeblefakie) [reply test/src/integration/integration_test.go]: Can you reply back to the comment?
I know you mentioned something in the daily meeting but it should be explained properly here.

So, my understanding `InitAndApply` and `RunTerraformCommand` use different contexts (like two different instances?) so it requires the init again?

File: test/src/integration/integration_test.go [hunk 4]
```diff
@@ -30,7 +30,6 @@ func TestEndToEndTerraform(t *testing.T) {
 		for _, m := range scalarModules {
 			terraformOptions := &terraform.Options{
 				TerraformDir: *terraformDir + *cloudProvider + "/" + m,
-				Vars:         map[string]interface{}{},
```
Comment (feeblefakie) [inline test/src/integration/integration_test.go:33]: Is this change related to this PR?
If not, please create another PR for that.

File: test/src/integration/integration_test.go (hunk 3 above)
Comment (MilanCorreggio) [reply test/src/integration/integration_test.go]: Actually after multiple test `init` is not needed.
I deleted the empty `Vars` optional object.
Goterratest is checking multiple option and by setting `Vars` to null it was reseting the instance.

File: test/src/integration/integration_test.go (hunk 4 above)
Comment (MilanCorreggio) [reply test/src/integration/integration_test.go]: I will thank you

File: test/src/integration/integration_test.go [hunk 5]
```diff
@@ -84,10 +84,11 @@ func TestEndToEndK8s(t *testing.T) {
...
```
Comment (feeblefakie) [inline test/src/integration/integration_test.go:87]: Sorry again, it should be deleted here?

**Review by feeblefakie — COMMENTED** [review]
Thank you for checking!
I still see some deletion that seems not related to this PR.
Please check again.

File: test/src/integration/integration_test.go (hunk 5 above)
Comment (MilanCorreggio) [reply test/src/integration/integration_test.go:87]: Yes it is in the k8s function :)

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
